src/system/sys_profiler.hh

```diff
@@ -65,15 +67,23 @@ class MPI_Functions {
 public:
     static int sum(int* val, MPI_Comm comm);
     static double sum(double* val, MPI_Comm comm);
+    static long sum(long* val, MPI_Comm comm);
+    
     static int min(int* val, MPI_Comm comm);
     static double min(double* val, MPI_Comm comm);
+    static long min(long* val, MPI_Comm comm);
+    
     static int max(int* val, MPI_Comm comm);
     static double max(double* val, MPI_Comm comm);
+    static long max(long* val, MPI_Comm comm);
 };
 
 // Assuming all compilers support constexpr
 #define CONSTEXPR_ constexpr
 
+// macro which does nothing but silence g++ compiler warning
+// about unused variable
+#define UNUSED(expr) (void)(expr)
```

Comment:
Remove. To deal with ierr return codes use cherr or cherr_assert from system/global_defs.hh instead.